app/jobs/concerns/import_job.rb

```diff
@@ -41,17 +41,7 @@ def validate_headers(filename)
 
   def import_records(filename)
     raise "No input file specified" unless filename
-    IOStreams.each_record(filename) do |record|
-      attrs = preprocess_attribute_values(translate_attribute_names(record))
-      initialized_model = category_model.new(
-          attrs.merge({processed_file_id: @processed_file.id, raw: false})
-      )
-      initialized_model.cleanse_data!
-      unless initialized_model.save
-        log("Error: Row #{stats[:row_count] + 2} is not valid. #{attrs}", :error)
-      end
-      increment_stats(attrs, initialized_model.persisted?)
-    end
+    @stats = CsvImporter.import(filename, category_model.name.underscore.humanize.titleize, @processed_file.id)
```

Comment:
Made CsvImporter responsible for doing the actual import. This makes the background jobs a lot dumber by deferring to the importer to do the actual process.